### PR TITLE
SONAR-7690 analysis uuid in table EVENTS

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PersistEventsStep.java#L82](https://github.com/SonarSource/sonarqube/blob/75d3da8070ededd5c80d6d43e19eff9c56382e67/server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistEventsStep.java#L82): Move this method into "PersistEventComponentVisitor". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3398)

File: server/sonar-web/src/main/webapp/WEB-INF/app/controllers/comparison_controller.rb [hunk 1]
```diff
@@ -69,20 +69,20 @@ def index
...
+      @versions = project.events.select { |event| event.category==EventCategory::KEY_VERSION && !suuids.include?(event.analysis_uuid.to_s) }
 
       # check if the latest snapshot if suggested or not (and if not, suggest it as "LATEST" => this is used for views or developers which do not have events)
-      latest_snapshot_id = resource.last_snapshot.id
-      current_and_suggested_sids = sids + @versions.map {|e| e.snapshot_id.to_s}
-      unless current_and_suggested_sids.include?(latest_snapshot_id.to_s)
-        @versions.unshift Event.new(:name => Api::Utils.message('comparison.version.latest'), :snapshot_id => latest_snapshot_id)
+      latest_snapshot_uuid = project.last_snapshot.uuid
+      current_and_suggested_suuids = suuids + @versions.map {|e| e.analysis_uuid.to_s}
+      unless current_and_suggested_suuids.include?(latest_snapshot_uuid.to_s)
```
Comment (simonbrandhof) [inline server/sonar-web/src/main/webapp/WEB-INF/app/controllers/comparison_controller.rb]: super minor, just for information, `to_s` is not required on the two previous lines as variables are strings.

**Comment by simonbrandhof** [comment]
Successfully tested. LGTM.

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PersistEventsStep.java#L82](https://github.com/SonarSource/sonarqube/blob/6cffa97b036ab628b91998cf294a6871a1de2804/server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistEventsStep.java#L82): Move this method into "PersistEventComponentVisitor". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3398)

**Pull request merged**